src/cartridges/int_adyen_overlay/cartridge/scripts/adyenSessions.js

```diff
@@ -36,35 +36,60 @@ function createSession(basket, customer, countryCode) {
       throw new Error('Could not do /sessions call');
     }
 
-    const paymentAmount = AdyenHelper.getCurrencyValueForApi(basket.getTotalGrossPrice());
-    const currencyCode = basket.currencyCode;
-
-    //TODO: Change AdyenHelper so that this object can have a different name. Not creating a payment request here
-    let paymentRequest = {
-      merchantAccount: AdyenHelper.getAdyenMerchantAccount(),
-      amount: {
-        currency: currencyCode,
-        value: paymentAmount.value,
-      },
-    };
-
-    // Add Risk data
-    if (AdyenHelper.getAdyenBasketFieldsEnabled()) {
-      paymentRequest.additionalData = RiskDataHelper.createBasketContentFields(
-          basket
-      );
+    let paymentAmount = 0;
+    let currencyCode = session.currency.currencyCode;
+
+    if(basket) {
+      paymentAmount = AdyenHelper.getCurrencyValueForApi(basket.getTotalGrossPrice());
+      currencyCode = basket.currencyCode;
     }
 
-    // L2/3 Data
-    if (AdyenHelper.getAdyenLevel23DataEnabled()) {
-      paymentRequest.additionalData = { ...paymentRequest.additionalData, ...adyenLevelTwoThreeData.getLineItems({Basket: basket}) };
+    let sessionsRequest = {};
+
+    // There is no basket for myAccount session requests
+    if(basket) {
+      //TODO: Change AdyenHelper so that this object can have a different name. Not creating a payment request here
+      let paymentRequest = {
```

Comment:
Can we use `sessionsRequest` here and extract the common fields like `merchantAccount`, `amount`?

Reply:
I deduped the basket checks in the class. But I would like to prevent variable reassigningif we only start using `sessionsRequest`.